Sources/MongoSwift/MongoDatabase.swift

```diff
@@ -272,6 +272,110 @@ public class MongoDatabase {
+      *   - `UserError.invalidArgumentError` if the `_id` field is projected out of the change stream documents by the
+      *     pipeline.
+      * - SeeAlso:
+      *   - https://docs.mongodb.com/manual/changeStreams/
+      *   - https://docs.mongodb.com/manual/meta/aggregation-quick-reference/
+      * - Note: Supported in MongoDB version 4.0 only.
+      */
+     public func watch(_ pipeline: [Document] = [],
+                       options: ChangeStreamOptions? = nil,
+                       session: ClientSession? = nil) throws ->
+                       ChangeStream<ChangeStreamDocument<Document>> {
+        return try self.watch(pipeline, options: options, session: session, withFullDocumentType: Document.self)
+     }
+
+     /**
+      * Starts a `ChangeStream` on a database. Excludes system collections. Associates the specified `Codable` type `T`
+      * with the `fullDocument` field in the `ChangeStreamDocument` emitted by the returned `ChangeStream`.
+      * - Parameters:
+      *   - pipeline: An array of aggregation pipeline stages to apply to the events returned by the change stream.
+      *   - options: An optional `ChangeStreamOptions` to use when constructing the `ChangeStream`.
+      *   - session: An optional `ClientSession` to use with this change stream.
+      *   - withFullDocumentType: The type that the change events emitted from the change stream will be decoded to.
+      * - Returns: A `ChangeStream` on all collections in a database.
+      * - Throws:
+      *   - `ServerError.commandError` if an error occurs on the server while creating the change stream.
+      *   - `ServerError.commandError` if the pipeline passed is invalid.
+      *   - `UserError.invalidArgumentError` if the options passed formed an invalid combination.
+      *   - `UserError.invalidArgumentError` if the `_id` field is projected out of the change stream documents by the
+      *     pipeline.
+      *   - `DecodingError` if an error occurs while decoding user-defined `withFullDocumentType` `Codable` type.
+      * - SeeAlso:
+      *   - https://docs.mongodb.com/manual/changeStreams/
+      *   - https://docs.mongodb.com/manual/meta/aggregation-quick-reference/
+      * - Note: Supported in MongoDB version 4.0 only.
+      */
+     public func watch<T: Codable>(_ pipeline: [Document] = [],
+                                   options: ChangeStreamOptions? = nil,
+                                   session: ClientSession? = nil,
+                                   withFullDocumentType: T.Type) throws ->
+                                   ChangeStream<ChangeStreamDocument<T>> {
+        let pipeline: Document = ["pipeline": pipeline]
+        let connection = try self._client.connectionPool.checkOut()
+        let opts = try encodeOptions(options: options, session: session)
+        return try self.withMongocDatabase(from: connection) { dbPtr in
```

Comment:
instead of duplicating this code, we can call through to `self.watch(pipeline, options: options, session: session, withReturnType: ChangeStreamDocument<T>)`. This also applies to the other `withFullDocumentType` variants